### DISCUSSION TITLE
DIAnn for Glycoproteomics

### DISCUSSION THREAD

**Discussion by MenaceGallagher** [vdemichev/DiaNN#562] [Q&A]
Hello Dr. Demichev,

I'm interested in using DIA-NN for glycoproteomics. Given the variability N- and O-linked glycan modifications, I am wondering whether DIA-NN is suitable for identifying glycopeptides. 

As it stands we have attempted to declare glycan modifications manually using the --var-mods, --full-unimod, --var-mod, and --monitor-mod commands paired with the respective fragments we wish to include in the FASTA digest.

Current issues we are facing:

- Inability to declare consensus sequence for PTM localization 
- Maximum of 5 variable modifications per run (this is a substantial limitation as there may be >100 possible masses)
- Declared modifications are reverted to UniMod terminology even when using --clear-mods command

Other questions I had related to the above were:
- You've identified in the documentation that the legacy predictor may be trained to identify a greater number of modifications, would this allow for glycan identification?
- Is there documentation available on the structure of the speclib format used by DIA-NN, and if so where can I find this?

Due to the large number of possible glycoforms that any given peptide may exist in, would it be possible to define a list of expected glycan modifications that are only considered where oxonium ions are detected?

Regards,
M. Gallagher

**Comment by vdemichev** [comment]
Hi M. Gallagher,

I would think the best way here would be to create a DDA-based spectral library. Cannot really comment on the ability of DIA-NN to identify glycans de-novo, we never tried anything like this. Probably quite challenging with DIA. 

A .speclib can be converted to .tsv using DIA-NN itself, just specify and output library file without specifying any raw data.

Best,
Vadim
